proc_CPMG_mw.py

```diff
@@ -532,20 +523,15 @@ def print_fun(x, f, accepted):
 
 for i,k in enumerate(power_list):
     temp = data['power',i].C
-    x = tE_axis 
-    ydata = temp.data.real
-    fl.next('DECAY DATA')
-    fl.plot(x,ydata, '.', alpha=0.4, label='%f'%k, human_units=False)
-    fitfunc = lambda p, x: p[0]*exp(-x/p[1])
-    errfunc = lambda p_arg, x_arg, y_arg: fitfunc(p_arg, x_arg) - y_arg
-    p0 = [11102.0,0.6]
-    p1, success = leastsq(errfunc, p0[:], args=(x, ydata))
-    x_fit = linspace(x.min(),x.max(),20000)
+    fl.plot(temp, '.', alpha=0.4, label='%f'%k, human_units=False)
+    temp = fitdata(temp)
+    A,R2,tau = symbols('A R2 tau')
+    data.symbolic_expression = A*sympy.exp(-tau*R2)
+    data.guess({A:11102.0,R2:0.6})
+    data.fit()
     fl.next('DECAY FITS')
-    fl.plot(x_fit, fitfunc(p1, x_fit),':', label='fit (T2 = %0.2f ms)'%(p1[1]*1e3), human_units=False)
-    xlabel('t (sec)')
-    ylabel('Intensity')
-    print("FOR POWER:",power_list[i],"W \tAMPLITUDE:",p1[0],"\tT2:",p1[1])
+    fl.plot(data.eval(400),':', label='fit (T2 = %0.2f ms)'%(p1[1]*1e3), human_units=False)
+    logger.info(strm("FOR POWER:",power_list[i],"W \tAMPLITUDE:",p1[0],"\tT2:",p1[1]))
     amplitude_r[i] = p1[0]
     T2_r[i] = p1[1]  
 
```

Comment:
have never used and don't know purpose of

Reply:
I am pretty sure it's to run DNP where you use CPMGs instead of single echoes or FIDs -- as I say above, I would add this to the to-do list, and deal with later.